### ISSUE TITLE
[Meta issue]: Rollups  Enhancements

### ISSUE THREAD

**Issue by praveensameneni** [opensearch-project/index-management#408]
Meta issue for Rollups enhancements:

## 1. [Count documents from source index](https://github.com/opensearch-project/index-management/issues/57)
- [x] Core change to use _doc_counts metadata field. https://github.com/opensearch-project/OpenSearch/pull/3985
- [X] Index management change to write doc counts to the metadata field https://github.com/opensearch-project/index-management/pull/439
## 2. [Allow a dynamic target index in ISM Rollups](https://github.com/opensearch-project/index-management/issues/61)
- [x] Add the ability to index to an alias which may be managed by a rollover policy
- [X] Write to a target index based on a mustache template https://github.com/opensearch-project/index-management/pull/435
## 3. [Allow rollup searches to multiple rollup indices if they have the same rollup](https://github.com/opensearch-project/index-management/issues/321)
- [X] https://github.com/opensearch-project/index-management/pull/440
## 4. [Allow query_string queries against rollups data](https://github.com/opensearch-project/index-management/issues/320) 
- [x] Enable searching with query string queries in the rollup interceptor

**Comment by downsrob** [comment]
For collecting the number of documents from the source index, the index management code base was prepped for a change in core which would make use of the _doc_counts metadata field to allow for seamlessly aggregating summary documents as we do in rollups. This core change did not come, but the change really belongs in core. This issue is open in the core repo here:
https://github.com/opensearch-project/OpenSearch/issues/3712
We should pick this up as the solution to aggregating document counts from summary docs.

**Comment by downsrob** [comment]
I have updated the individual issues, but to summarize the progress at the time of the 2.2 release:
1. Doc counts changes are fully complete
2. Mustache field templates based on source index name is done. Using an alias as a target index will be targeted for the next release. There are some user experience questions related to that feature which I have commented in the issue.
3. Searching multiple rollup indices is fully implemented. Integration and unit tests have not yet been added and should be added as soon as possible.
4. There is an open PR for query string queries, but this PR does not include support for translating field names in the query string or the default field value. It is going to take more investigation to figure out how to best parse and then translate the query string.
